File: frontend/src/api/geoApi.js
import axiosClient from './axiosClient';

export async function fetchCatalogCities() {
  const { data } = await axiosClient.get('/geo/catalog-cities');
  if (!data.success) throw new Error(data.message || 'Không tải được danh sách thành phố');
  return data.data;
}

export async function fetchProvinces() {
  const { data } = await axiosClient.get('/geo/provinces');
  if (!data.success) throw new Error(data.message || 'Không tải được tỉnh/thành');
  return data.data;
}

export async function fetchGeoBundle() {
  const { data } = await axiosClient.get('/geo/bundle');
  if (!data.success) throw new Error(data.message || 'Không tải được dữ liệu địa giới');
  return data.data;
}

File: mobile/lib/api/geo_api.dart
import 'api_client.dart';

class GeoApi {
  GeoApi(this._client);

  final ApiClient _client;

  Future<List<String>> fetchCatalogCities() async {
    final data = await _client.getJson('/geo/catalog-cities');
    final list = data['data'];
    if (list is! List) return [];
    return list.map((e) => e.toString()).where((s) => s.isNotEmpty).toList();
  }

  Future<List<Map<String, dynamic>>> fetchProvinces() async {
    final data = await _client.getJson('/geo/provinces');
    final list = data['data'];
    if (list is! List) return [];
    return list.map((e) => Map<String, dynamic>.from(e as Map)).toList();
  }
}
